core/xchain/entitlement/entitlement.go

```diff
@@ -26,6 +26,21 @@ func (e *Evaluator) EvaluateRuleData(
 	return e.evaluateOp(ctx, opTree, linkedWallets)
 }
 
+func isNilOrCancelled(err error) bool {
+	return err == nil || err == context.Canceled
```

Comment:
Consider using `errors.Is(err, context.Cancelled)`. This also verifies is err is wrapped around a canceled err which is probably what you want to check for.

Reply:
Nice, thank you!